simplehash/schemav2.go

```diff
@@ -38,10 +39,20 @@ type HashOptions struct {
 	asConfirmed            bool
 	prefix                 []byte
 	committed              *timestamppb.Timestamp
+	idcommitted            []byte
 }
 
 type HashOption func(*HashOptions)
 
+// WithIDCommitted includes the snowflakeid unique commitment timestamp in the hash
+// idcommitted is never (legitimately) zero
+func WithIDCommitted(idcommitted uint64) HashOption {
```

Comment:
We need to bump schema version for adding a new field?

I think i would prefer we have 2 schema versions, 1 with the `idcomitted` required and 1 without `idcommitted`.

Having optionality within the schema version may cause a headache down the line, i.e it could be one of these x hashes...

Reply:
I don't think that is necessary. This does not change the schema for simplehash. It is an accomodation for forestrie hashing. The idcommitted value is not explicitly in the event data at all

Reply:
as discussed, moved the options to a new file options.go to avoid ambiguity about what the schema is